src/lex/cpp.rs

```diff
@@ -886,6 +886,15 @@ impl<'a> PreProcessor<'a> {
             this.tokens_until_newline()
                 .into_iter()
                 .map(|res| res.map(|loc| loc.data))
+                .enumerate()
+                .map(|(i, x)| {
+                    if i == 0 {
+                        vec![x]
+                    } else {
+                        vec![x, Ok(Token::Whitespace(String::from(" ")))]
+                    }
+                })
+                .flatten()
```

Comment:
Why do you add these spaces here in the define? Does `tokens_until_newline` not return whitespace tokens?

Reply:
`tokens_until_newline` (at the moment) does not include whitespace tokens. I did not want to change it since it is somewhat separate from the preprocessor. I can try adding whitespace tokens to it and see what happens now that it is in a stable state. I did notice that `clang` only puts one space when replacing preprocessor defines, regardless of the original spacing.

Reply:
Hmm interesting ... I suppose since the behavior is correct we can try to go back and improve the spacing later.

Reply:
Also, please add a comment to this effect either here or at `tokens_until_newline`.

Reply:
I actually went back and did it the proper way by changing tokens_until_newline. Unfortunately I had to rework some stuff for boolean_expr because whitespace show up in the replacement stage. I'll push as soon as I double check the tests. Also, I think the rework will be much less of an eyesore.